addons/html_editor/static/src/core/delete_plugin.js

```diff
@@ -910,15 +913,7 @@ export class DeletePlugin extends Plugin {
         const isNonEditable = (node) => !isContentEditable(node);
         const startUneditable = findFurthest(startContainer, commonAncestor, isNonEditable);
         if (startUneditable) {
-            // @todo @phoenix: Review this spec. I suggest this instead (no block merge after removing):
-            // startContainer = startUneditable.parentElement;
-            // startOffset = childNodeIndex(startUneditable);
-            const leaf = previousLeaf(startUneditable);
-            if (leaf) {
-                range.setStart(leaf, nodeSize(leaf));
-            } else {
-                range.setStart(commonAncestor, 0);
-            }
```

Comment:
@Zynton This was the reason for the whole body getting deleted in one of the tests in banner.test.js